docs/spec_draft.md

```diff
@@ -1526,6 +1527,64 @@ operations correspond to [stablehlo.minimum](#stablehlominimum) and
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.collective_permute
+
+### Semantics
+
+Within each process group in the StableHLO grid, sends and receives data between
+processes in the group.
+
+The operation splits the StableHLO grid into `process_groups` as follows:
+  * `channel_id <= 0`,
+    `cross_replica(replica_groups)`.
+  * `channel_id > 0`,
+    `cross_partition(replica_groups)`.
+
+Afterwards, `result@receiver` is given by
+ * `operand@process_groups[i][0]`, if there exist an `i` such that  `process_groups[i][1] = receiver`.
```

Comment:
instead of using receiver we can use `source` and `target` because that is the terminology of the attribute `source_target_pairs`.

Reply:
also can use `source_target_pairs` instead of `process_groups`

Reply:
>> instead of using receiver we can use source and target because that is the terminology of the attribute source_target_pairs.

Done.

>> also can use source_target_pairs instead of `process_groups`

`process_groups` is the groups of process pairs formed depending on the group forming strategy `cross_replica` or `cross_partition`. We cannot use the `source_target_pairs` directly as it has replica  or partition ids, but  not the process ids which is what we need to  for the communication.